drivers/volume/portworx/portworx.go

```diff
@@ -1091,6 +1091,46 @@ func (d *portworx) GetClusterPairingInfo() (map[string]string, error) {
 	return pairInfo, nil
 }
 
+func (d *portworx) DecommissionNode(n node.Node) error {
+
+	if err := d.StopDriver([]node.Node{n}, false); err != nil {
+		return &ErrFailedToDecommissionNode {
+			Node:    n.Name,
+			Cause:   fmt.Sprintf("Failed to stop driver on node: %v. Err: %v", n.Name, err),
+		}
+	}
+	clusterManager := d.getClusterManager()
+	pxNode, err := clusterManager.Inspect(n.VolDriverNodeID)
+	if err != nil {
+		return &ErrFailedToDecommissionNode {
+			Node:    n.Name,
+			Cause:   fmt.Sprintf("Failed to inspect node: %v. Err: %v", pxNode, err),
+		}
+	}
+
+	if err = clusterManager.Remove([]api.Node{pxNode}, false); err != nil {
+		return &ErrFailedToDecommissionNode {
+			Node:    n.Name,
+			Cause:   err.Error(),
+		}
+	}
+	return nil
+}
+
+func (d *portworx) DecommissionNodeStatus(n node.Node) (string, error) {
```

Comment:
This function isn't very clear on that it does.

Rather just have something generic like `GetNodeStatus`. Just have the interface return statuses of type api.Status and error. api.Status is okay to return since that is openstorage and not Portworx specific.